sentry-kotlin-multiplatform/src/commonAppleMain/kotlin/io/sentry/kotlin/multiplatform/extensions/SentryOptionsExtensions.kt

```diff
@@ -28,11 +30,29 @@ internal fun CocoaSentryOptions.applyCocoaBaseOptions(options: SentryOptions) {
     this.enableAutoSessionTracking = options.enableAutoSessionTracking
     this.beforeSend = { event ->
         dropKotlinCrashEvent(event as NSExceptionSentryEvent?) as SentryEvent?
+
+        val cocoaName = BuildKonfig.SENTRY_COCOA_PACKAGE_NAME
+        val cocoaVersion = BuildKonfig.SENTRY_COCOA_VERSION
+        options.sdk.apply {
+            val names = this.packages?.map { it.name }
+            if (names?.contains(cocoaName) != true) {
+                this.addPackage(cocoaName, cocoaVersion)
+            }
+        }
+        val sdk = event?.sdk?.toMutableMap()
+        sdk?.set(
+            "packages",
+            options.sdk.packages?.map {
+                mapOf("name" to it.name, "version" to it.version)
+            }
+        )
+        event?.sdk = sdk
```

Comment:
there's no way to attach packages in the cocoa SDK via PrivateSentrySDKOnly, so it is currently done in the beforeSend hook. Although I'm not sure if this is a reliable way to attach them, any thoughts on this?

Reply:
I think `beforeSend` is the only place that is possible to do this now. But what happens if the user set a `beforeSend` callback?

Reply:
Right now the kmp sdk doesn't have support for a public facing `beforeSend` hook but it's possible to add these packages regardless of what the user sets in their `beforeSend`

Reply:
Packages need to be added in beforeSend, AFAIK there's no other way yet https://github.com/getsentry/sentry-dart/blob/11fb40841ad3d6fc46e02eb8adeb74d5b6932582/flutter/ios/Classes/SentryFlutterPluginApple.swift#L233-L250
https://github.com/getsentry/sentry-cocoa/issues/2270